Add atoms charts aa1 to implement 1

pinterid opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently there is no 2D calendar chart. #17 
- Currently there is no 3D calendar chart. #18 
- Currently there is no contribution radar chart. #19 
- Currently there is no latest activity chart. #20 


**What is the new behavior (if this is a feature change)?**
- Now there is a 2D calendar chart. #17 
- Now there is a 3D calendar chart. #18 
- Now there is a contribution radar chart. #19 
- Now there is a latest activity chart. #20 


**Other information**:
- Ref: 🐍

schettn committed:
Add Calendar2D Component

The chart for the 2D calendar has been added.
It is not displayed yet.

schettn committed:
Add Calendar3D Component

The chart for the 3D calendar has been added.
It is not displayed yet.

schettn committed:
Add ContribRadar Component

The chart for the contribution radar
has been added. It is not displayed yet.

schettn committed:
Add LatestActivity Component

The chart for the latest activity
has been added. It is not displayed yet.

schettn committed:
Add Atom Descriptions

Descriptions for the usage of all atoms
have been added.

schettn committed:
Adjust newlines

There are only newlines to separate internal
and external imports in the import section now.

pinterid commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+/** @class A two dimensional calendar which displays each days contributions */
+class Calender2D extends React.Component {
+  constructor(props) {
+    super(props);
```
Add newline.

schettn commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+  constructor(props) {
+    super(props);
+    this.myInput = React.createRef();
+
```
```suggestion
    super(props);
    
    this.myInput = React.createRef();
```

schettn commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+  componentDidMount = () => {
+    // Fill calendar
+    this.setCalendar(this.props);
+
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+    // Add resize listener
+    window.addEventListener("resize", this.updateDimensions);
+
+    // We could implement color RGB cycling here
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+    let month = new Array();
+
+    // Get current month
+    let current;
```
```suggestion
    let current;
    
```

schettn commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+  };
+
+  displayDailyInfo = (day, wkey, dkey) => {
+    let cname = "item-" + wkey + "-" + dkey;
```
```suggestion
    let cname = "item-" + wkey + "-" + dkey;
    
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+//#region > Components
+/**
+ * @class A three dimensional calendar which displays each days contributions,
+ *        and contribution related statistics e.g. busiest days and streaks
```
```suggestion
 *        and contribution related statistics e.g. busiest days and streaks.
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+    super(props);
+
+    this.myInput = React.createRef();
+
```
```suggestion
```

pinterid commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+  componentDidMount = () => {
+    if (this.props.platformData) {
+      // Add resize listener
+      window.addEventListener("resize", this.updateDimensions);
```
Add newline.
```suggestion
      window.addEventListener("resize", this.updateDimensions);
      
      this.setState(
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+    // Use caching for year change
+    //console.log(this.props.year);
+    //console.log(nextProps.year);
+  };
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+  };
+
+  renderTopStats() {
+    let countTotal, averageCount, datesTotal, maxCount, dateBest, contribData;
```
```suggestion
    let countTotal, averageCount, datesTotal, maxCount, dateBest, contribData;

```

pinterid replied:
Add newline.
```suggestion
    let countTotal, averageCount, datesTotal, maxCount, dateBest, contribData;
    
    if (this.props.year) {
```

pinterid commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+    window.removeEventListener("resize", this.updateDimensions);
+  }
+
+  componentWillReceiveProps = (nextProps) => {
```
Remove it, it does nothing.

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+    datesTotal =
+      moment(contributionCalendar.startDate).format("MMM DD, YYYY") +
+      " - " +
+      moment(contributionCalendar.endDate).format("MMM DD, YYYY");
```
```suggestion
    averageCount =
      Math.round(
        (contribData.contributions.total / 365 + Number.EPSILON) * 100
      ) / 100;
      
    datesTotal =
      moment(contributionCalendar.startDate).format("MMM DD, YYYY") +
      " - " +
      moment(contributionCalendar.endDate).format("MMM DD, YYYY");
      
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+  }
+
+  renderBottomStats() {
+    let streakLongest, datesLongest, streakCurrent, datesCurrent, contribData;
```
```suggestion
    let streakLongest, datesLongest, streakCurrent, datesCurrent, contribData;
    
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+    </span>\n
+    </span>\n
+    </div>`;
+    return { __html: html };
```
Rewrite. You don't need a extra variable here. Move this to the export object.

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+    </span>\n
+    </span>\n
+    </div>`;
+    return { __html: html };
```
Rewrite. You don't need a extra variable here. Move this to the export object.

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+  renderIsometrics = () => {
+    if (this.context) {
+      const obelisk = require("obelisk.js");
+
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+
+      // Create a canvas 2D point for pixel view world
+      let point = new obelisk.Point(70, 70);
+
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+
+      // Create view instance to nest everything
+      // Canvas could be either DOM or jQuery element
+      let pixelView = new obelisk.PixelView(this.context, point);
```
```suggestion
      let pixelView = new obelisk.PixelView(this.context, point);
      
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      pixelView.clear();
+
+      // Get contributions of the selected year
+      let contribData;
```
```suggestion
      let contribData;
      
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      let SIZE = 2 * Math.round(this.state.width / 80 / 2);
+      if (SIZE <= 8) {
+        SIZE = 8;
+      }
```
```suggestion
      let SIZE = 2 * Math.round(this.state.width / 80 / 2);
      
      if (SIZE <= 8) {
        SIZE = 8;
      }
      
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      if (SIZE <= 8) {
+        SIZE = 8;
+      }
+      let MAXHEIGHT = 100;
```
You have two options.
1. Switch to lowercase and const
2. Declare the variable as const and move it out of the class.

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      let contributions = contribData.calendar;
+
+      // Define basic variables
+      let SIZE = 2 * Math.round(this.state.width / 80 / 2);
```
You have two options.
1. Switch to lowercase and const
2. Declare the variable as const and move it out of the class.

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      let x = 0;
+      let y = 0;
+      let maxCount = 0; // Max number of contributions / day in the last year
+
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      let values = [];
+
+      contributions.weeks.map((week, wkey) => {
+        values[wkey] = [];
```
```suggestion
        values[wkey] = [];
        
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+      values.map((week, wi) => {
+        week.map((day, di) => {
+          // Normalize the values to achieve even distribution
+          let cubeHeight = 3;
```
```suggestion
          let cubeHeight = 3;
          
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,372 @@
+  checkCache = () => {
+    const cache = localStorage.getItem("3dChart");
+    if (cache) {
+      const cacheObject = JSON.parse(cache);
```
```suggestion
      const cacheObject = JSON.parse(cache);
      
```

schettn reviewed (changes requested):
I left some suggestions.

schettn committed:
Apply suggestions from code review

Most of the suggestions have been applied.
All other comments will be resolved imidiately.

Co-authored-by: Pinterics David <55298934+pinterid@users.noreply.github.com>

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -66,6 +61,8 @@ class Calendar3D extends React.Component {
   renderTopStats() {
     let countTotal, averageCount, datesTotal, maxCount, dateBest, contribData;
+    
+    if (this.props.year) {
```
```suggestion
```

schettn commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -308,6 +311,7 @@ class Calendar3D extends React.Component {
     const cache = localStorage.getItem("3dChart");
```
```suggestion
    const cache = localStorage.getItem("3dChart");
    
```

schettn commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,243 @@
+  getEachMonth = (pos) => {
+    // Create new empty array
+    let month = new Array();
+
```
```suggestion
```

schettn reviewed (changes requested):
There are some errors produced by the previous commit

schettn committed:
Apply suggestions from code review

All other suggestions are implemented now.

schettn committed:
Improve code quality

The quality of the code has been improved.

pinterid committed:
Improve code quality

The code quality was improved due to a request by Codacy.

schettn reviewed (approved):
🚀

pinterid commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,240 @@
+import "./calendar2d.scss";
+//#endregion
+
+const months = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
```
Add a region and change "months" to "Months".
```suggestion
//#region > Constant Variables
const Months = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
//#endregion
```

schettn replied:
The regions are correct but do not change the variable name with a suggestions. This would crash the whole code.
Btw, it is "MONTHS"
```suggestion
//#region > Constant Variables
const months = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
//#endregion
```

pinterid commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -0,0 +1,186 @@
+  };
+
+  componentDidMount = () => {
+    // First render chart with current year
```
```suggestion
    // Start of by rendering chart with current year
```

pinterid commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -0,0 +1,186 @@
+  }
+
+  fillChart = (results) => {
+    console.log(results);
```
Add newline or remove console.log.
```suggestion
    console.log(results);
    
```

pinterid commented on src/components/atoms/charts/LatestActivity/index.jsx:
```diff
@@ -0,0 +1,170 @@
+  };
+
+  componentDidMount = () => {
+    // First render chart with current week
```
```suggestion
    // Start of by rendering chart with current week
```

pinterid commented on src/components/atoms/charts/LatestActivity/index.jsx:
```diff
@@ -0,0 +1,170 @@
+
+  calculateSources = (year, activity) => {
+    const { statistic } = this.props;
+
```
Remove newline.
```suggestion
```

pinterid commented on src/components/atoms/index.js:
```diff
@@ -1,11 +1,15 @@
 //> Atoms
 // Import all components to export them for easy access from parent components
-//import Atom from "./Atom";
+//>> User profile
```
```suggestion
//> User profile
```

pinterid reviewed (commented):
Make some changes.

schettn committed:
Apply suggestions from code review

All suggestions are implemented now.

Co-authored-by: Pinterics David <55298934+pinterid@users.noreply.github.com>

schettn committed:
Update src/components/atoms/index.js

Now it is a valid heading.

Co-authored-by: Pinterics David <55298934+pinterid@users.noreply.github.com>

schettn committed:
Improve code quality

The naming of a constant variable
has been refined.

pinterid commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -0,0 +1,242 @@
+          <MDBRow className="text-center">
+            {MONTHS.map((month, key) => {
+              return (
+                <MDBCol size="MONTHS" key={key}>
```
```suggestion
                <MDBCol size="months" key={key}>
```

pinterid reviewed (commented):
Looks good after a change.

pinterid committed:
Improve code quality

The code quality has been improved due to a request by @pinterid.

pinterid reviewed (commented):
Looks good.

kleberbaum commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,374 @@
+
+  renderIsometrics = () => {
+    if (this.context) {
+      const obelisk = require("obelisk.js");
```
Please add a newline.

kleberbaum commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,374 @@
+        size = 8;
+      }
+
+      const maxHight = 100;
```
Please add a newline.

kleberbaum commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -0,0 +1,374 @@
+  renderCache = () => {
+    const data = localStorage.getItem("3dChart");
+    const dataObject = JSON.parse(data);
+    const context = this.context.getContext("2d");
```
Please add a newline.

kleberbaum commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -0,0 +1,187 @@
+  };
+
+  calculateSources = (nextPropsYear) => {
+    const { statistic } = this.props;
```
Please add a newline.

kleberbaum commented on src/components/atoms/charts/LatestActivity/index.jsx:
```diff
@@ -0,0 +1,169 @@
+  };
+
+  calculateSources = (year, activity) => {
+    const { statistic } = this.props;
```
Please add a newline.

kleberbaum commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -0,0 +1,187 @@
+  calculateSources = (nextPropsYear) => {
+    const { statistic } = this.props;
+    let totalReviews = 0,
+      totalIssues = 0,
```
Please fix indentation.

kleberbaum commented on src/components/atoms/charts/LatestActivity/index.jsx:
```diff
@@ -0,0 +1,169 @@
+  }
+
+  fillChart = (results, lastWeek) => {
+    console.log(results, lastWeek);
```
Please add a anker.

kleberbaum commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -0,0 +1,187 @@
+  }
+
+  fillChart = (results) => {
+    console.log(results);
```
Please add a anker.

kleberbaum reviewed (changes requested):
Should be ok to merge after a few changes.

schettn committed:
Improve code quality

The code quality has been improved
due to a request of @kleberbaum.

schettn committed:
Improve code quality

Comment out all console.log

kleberbaum reviewed (approved):
Should be ok to merge.
